moped-editor/src/views/projects/projectView/ProjectComponents/featureUtils.js

```diff
@@ -0,0 +1,98 @@
+import { v4 as uuidv4 } from "uuid";
+import { getIntersectionLabel } from "./utils";
+
+/**
+ * Captured from feature object example
+ * {
+ *    "geometry": {
+ *        "type": "LineString",
+ *        "coordinates": [],
+ *    },
+ *    "properties": {
+ *        "OBJECTID": 20041,
+ *        "CTN_SEGMENT_ID": 153083,
+ *        "FULL_STREET_NAME": "W MARTIN LUTHER KING JR BLVD",
+ *        "FROM_ADDRESS_MIN": 300,
+ *        "TO_ADDRESS_MAX": 303,
+ *        "LINE_TYPE": "On-Street",
+ *        "GLOBALID": "cb89d800-d6a3-491f-91e6-61cab17ac738",
+ *        "Shape__Length": 60.3716805891339,
+ *        "id": 20041,
+ *        "_layerId": "ctn-lines-underlay",
+ *        "_label": "300 BLK W MARTIN LUTHER KING JR BLVD"
+ *    }
+ * }
+ */
+
+/**
+ * Make a feature object from a clicked CTN feature and AGOL returned data about it
+ * @param {Object} featureFromAgolGeojson - a feature object from AGOL
+ * @param {Object} clickedFeature - a feature object from a Mapbox click event
+ * @param {Object} ctnLinesGeojson - a feature collection of CTN lines currently in map view
+ * @returns {Object} - a feature object
+ */
+export const makeCapturedFromLayerFeature = (
+  featureFromAgolGeojson,
+  clickedFeature,
+  ctnLinesGeojson
+) => {
+  const newFeature = {
+    geometry: featureFromAgolGeojson.geometry,
+    properties: {
+      ...featureFromAgolGeojson.properties,
+      id: featureFromAgolGeojson.id,
+      // AGOL data doesn't include layer so we grab it from the clicked Mapbox feature
+      _layerId: clickedFeature.layer.id,
+    },
+  };
+
+  if (newFeature.properties._layerId.includes("point")) {
+    newFeature.properties._label = getIntersectionLabel(
+      newFeature,
+      ctnLinesGeojson
+    );
+  } else {
+    newFeature.properties._label = `${newFeature.properties.FROM_ADDRESS_MIN} BLK ${newFeature.properties.FULL_STREET_NAME}`;
+  }
+
+  return newFeature;
+};
+
+/**
+ * Drawn feature object example
+ * {
+ *   "id": "9e0c688733283be4991cb9df39ae284d",
+ *   "geometry": { coordinates: [], type: "LineString" },
+ *   "type": "Feature",
+ *   "properties": {
+ *     "DRAW_ID": "88230580-099f-4c93-b8e5-ad3086caf0e6",
+ *     "_layerId": "drawnByUserLine",
+ * }
+ */
```

Comment:
these examples are super helpful

Reply:
i'm lazily not looking at Frank's branch, but i'm curious how this will shape into the new layer schema. i'm sure you're working through it!

Reply:
For sure, it is definitely getting interesting! So far, I'm trying to focus on making the munging well-defined going in and out of the map. It will probably look similar to this in the end. 🤞